File: 프로그래머스/2/76502. 괄호 회전하기/괄호 회전하기.js
function check(s) {
    const stack = [s[0]];
    
    for (let i = 1; i < s.length; i++) {
        const c = s[i];
        const h = stack.at(-1);
        if (h === '(' && c === ')') stack.pop();
        else if (h === '[' && c === ']') stack.pop();
        else if (h === '{' && c === '}') stack.pop();
        else stack.push(c);
    }
    
    return stack.length === 0;
}

function solution(s) {
    let answer = 0;
    
    for (let i = 0; i < s.length; i++) {
        const newString = s.slice(i) + s.slice(0, i);
        if (check(newString)) answer++;
    }
    
    return answer;
}

File: 프로그래머스/2/76502. 괄호 회전하기/괄호 회전하기.py
def check(s):
    stack = ['', s[0]]
    
    for c in s[1:]:
        h = stack[-1]
        if h == '(' and c == ')': stack.pop()
        elif h == '[' and c == ']': stack.pop()
        elif h == '{' and c == '}': stack.pop()
        else: stack.append(c)
    
    return len(stack) == 1

def solution(s):
    answer = 0
    n = len(s)
    
    for i in range(n):
        new_string = s[i:] + s[:i]
        if check(new_string):
            answer += 1
            
    return answer
